pulpcore/cli/common/generic.py

```diff
@@ -1002,29 +1000,30 @@ def version_callback(
         )
         return repo_ver_ctx
 
-    @click.command("list")
-    @click.option("-t", "--type", "type", type=click.Choice(names), default=names[0])
+    @pulp_command("list")
+    @click.option("--all-types", is_flag=True)
     @limit_option
     @offset_option
     @repository_option
     @click.option("--version", type=int, callback=version_callback)
     @pass_pulp_context
+    @click.pass_context
     def content_list(
+        ctx: click.Context,
         pulp_ctx: PulpContext,
         version: PulpRepositoryVersionContext,
         offset: Optional[int],
         limit: Optional[int],
-        type: Optional[str],
+        all_types: Optional[bool],
         **params: Any,
     ) -> None:
         parameters = {k: v for k, v in params.items() if v is not None}
         parameters.update({"repository_version": version.pulp_href})
-        result = content_contexts[type](pulp_ctx).list(
-            limit=limit, offset=offset, parameters=parameters
-        )
+        ctx_obj = PulpContentContext(pulp_ctx) if all_types else ctx.obj
```

Comment:
So this will either result in an object of type `PulpContentContext` or `PulpRepositroy`? That sounds wrong to me.
ALso what is the desired behaviour if `all_types` is `False`?

Reply:
No it will always be a subtype of `PulpContentContext`, I should probably use `pass_content_context` since this group's context object will always be set to a content context due to the type option.